ethereum/contracts/ERC20App.sol

```diff
@@ -73,20 +79,39 @@ contract ERC20App is AccessControl {
             "Invalid channel ID"
         );
 
+        uint8 _decimals;
+        string memory _name;
+        string memory _symbol;
+
+        (_name, _symbol, _decimals) = tokenDetails(_token);
+
         balances[_token] = balances[_token] + _amount;
 
         emit Locked(_token, msg.sender, _recipient, _amount, _paraId);
 
+        OutboundChannel channel = OutboundChannel(
+            channels[_channelId].outbound
+        );
+
+        bytes memory createCall;
+        if (!wrappedTokenList[_token]) {
+            createCall = encodeCreateTokenCall(
```

Comment:
should we move all this logic within the if statement into its own separate function?